File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>

int alphacount(string txt);
int wordcount(string txt);
int sentcount(string txt);

// Determines the reading level of the text provided by the user
int main(void)
{
    string text = get_string("Text: ");
    int letters = alphacount(text);
    int words = wordcount(text);
    int sentences = sentcount(text);

    float l = (letters * 100) / (float) words;
    float s = (sentences * 100) / (float) words;
    double index = (0.0588 * (double) l) - (0.296 * s) - 15.8;

    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", (int) round(index));
    }
}

// Counts letters
int alphacount(string txt)
{
    int l = 0;
    for (int i = 0; txt[i] != '\0'; i++)
    {
        if (isalpha(txt[i]))
        {
            l++;
        }
    }
    return l;
}

// Counts words
int wordcount(string txt)
{
    int w = 1;
    for (int i = 0; txt[i] != '\0'; i++)
    {
        if (isspace(txt[i]))
        {
            w++;
        }
    }
    return w;
}

// Counts sentences
int sentcount(string txt)
{
    int s = 0;
    for (int i = 0; txt[i] != '\0'; i++)
    {
        if (txt[i] == '!' || txt[i] == '.' || txt[i] == '?')
        {
            s++;
        }
    }
    return s;
}

File: readability.py
from cs50 import get_string

def alphacount(txt):
    l = 0
    i = 0
    for i in range(0, len(txt)):
        if txt[i].isalpha():
            l += 1
    return l
    
def wordcount(txt):
    w = 1
    i = 1
    for i in range(0, len(txt)):
        if txt[i].isspace():
            w += 1
    return w

def sentcount(txt):
    s = 0
    for i in range(0, len(txt)):
        if (txt[i] == '!' or txt[i] == '.' or txt[i] == '?'):
            s += 1
    return s

def main():
    text = get_string("Text: ")
    letters = alphacount(text)
    words = wordcount(text)
    sentences = sentcount(text)
    
    print(letters)
    print(words)
    print(sentences)
    
    l = (letters * 100) / words
    s = (sentences * 100) / words
    index = (0.0588 * l) - (0.296 * s) - 15.8
    
    print(index)
    
    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {int(round(index))}")
        
if __name__ == "__main__":
    main()